### PR TITLE
Add missing options

### PR DESCRIPTION
**Pull request by francisdaigle**
Add missing plugin options to TS declaration file.

### CODE REVIEW

**Comment by hsynlms** [comment]
Hi @francisdaigle 

Thank you for the PR!

I just wonder that why you did some changes in `index.test-d.ts`?

File: .gitignore [hunk 1]
```diff
@@ -103,5 +103,9 @@ dist
 # TernJS port file
 .tern-port
 
+# vscode
+.vscode
+*code-workspace
+
```
Comment (hsynlms) [inline .gitignore]: I think we do not need these changes, It depends on people's environment setup and may vary.

**Comment by francisdaigle** [comment]
Hi @hsynlms 

The missing options were added to the [TSD](https://github.com/SamVerschueren/tsd) test as well.

File: .gitignore (hunk 1 above)
Comment (francisdaigle) [reply .gitignore]: Hmmm... alright, but is that not what `.gitignore` is for? I believe you will find similar lines in [Fastify's `.gitignore`](https://github.com/fastify/fastify/blob/main/.gitignore#L129).

**Comment by hsynlms** [comment]
I mean lines 32 and 33 change nothing indeed. Also, role, scope, and request properties assignments (lines 17, 18, 19) are useless unless there is no user object within Fastify request.

**Comment by francisdaigle** [comment]
> I mean lines 32 and 33 change nothing indeed. 

Just being more explicit. 

> Also, role, scope, and user properties assignments (lines 17, 18, 19) are useless unless there is no user object within Fastify request.

Hmmm... to my knowledge, this is the standard way of going about things. I believe the point is to register the plugin and test that it compiles. Here's an [example](https://github.com/fastify/fastify-mongodb/blob/master/index.test-d.ts).

File: .gitignore (hunk 1 above)
Comment (hsynlms) [reply .gitignore]: You are right but in my opinion, environment-specific things should not be added to gitignore file. If it's project-specific things then good to go.

Comment (hsynlms) [reply .gitignore]: I will revise the gitignore file accordingly. You need to ignore environment-specific files in your global gitignore.

**Review by hsynlms — CHANGES_REQUESTED** [review]
gitignore file revertion

**Review by hsynlms — APPROVED** [review]

**Comment by hsynlms** [comment]
Thank you @francisdaigle

**Pull request merged**